File: src/main/java/net/softbell/bsh/domain/converter/TriggerStatusRuleConverter.java
package net.softbell.bsh.domain.converter;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

import net.softbell.bsh.domain.TriggerStatusRule;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 인증 상태 자료형 DB 컨버터
 */
@Converter(autoApply = true)
public class TriggerStatusRuleConverter implements AttributeConverter<TriggerStatusRule, Integer>
{
	@Override
	public Integer convertToDatabaseColumn(TriggerStatusRule triggerStatusRule)
	{
		if (triggerStatusRule == null)
			return null;
		
		return triggerStatusRule.getCode();
	}

	@Override
	public TriggerStatusRule convertToEntityAttribute(Integer code)
	{
		return TriggerStatusRule.ofLegacyCode(code);
	}
}

File: src/main/kotlin/net/softbell/bsh/domain/converter/TriggerStatusRuleConverter.kt
package net.softbell.bsh.domain.converter

import net.softbell.bsh.domain.TriggerStatusRule
import javax.persistence.AttributeConverter
import javax.persistence.Converter

/**
 * @author : Bell(bell@softbell.net)
 * @description : 인증 상태 자료형 DB 컨버터
 */
@Converter(autoApply = true)
class TriggerStatusRuleConverter : AttributeConverter<TriggerStatusRule, Int> {
    override fun convertToDatabaseColumn(triggerStatusRule: TriggerStatusRule): Int {
        return triggerStatusRule.code
    }

    override fun convertToEntityAttribute(code: Int): TriggerStatusRule? {
        return TriggerStatusRule.ofLegacyCode(code)
    }
}